drivers/platform/nintendo3ds/ctr_gpio.c

```diff
@@ -0,0 +1,225 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ *  ctr_gpio.c
+ *
+ *  Copyright (C) 2021 Santiago Herrera
+ */
+
+#define DRIVER_NAME "3ds-gpio"
+#define pr_fmt(fmt)	DRIVER_NAME ": " fmt
+
+#include <linux/io.h>
+#include <linux/irq.h>
+#include <linux/init.h>
+#include <linux/slab.h>
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/interrupt.h>
+#include <linux/gpio/driver.h>
+#include <linux/platform_device.h>
+#include <linux/mod_devicetable.h>
+#include <linux/irqchip/chained_irq.h>
+
+struct ctr_gpio {
+	struct device *dev;
+
+	unsigned ngpios;
+	void __iomem *dat;
+	void __iomem *dir;
```

Comment:
`dir` can be local to `ctr_gpiointc_probe`?

Reply:
Probably? I find it might be useful at some point to keep ahold of the direction register base.